src/status_im2/contexts/wallet/common/sheets/network_preferences/view.cljs

```diff
@@ -1,70 +1,92 @@
+                                                                          network-name)
+                                                             (swap! network-preference-names-state disj
+                                                               network-name)
+                                                             (swap! network-preference-names-state conj
+                                                               network-name)))]
+    (fn [{:keys [on-save theme]}]
+      (let [network-details  (rf/sub [:wallet/network-details])
+            mainnet          (first network-details)
+            layer-2-networks (rest network-details)
+            current-networks (->> network-details
+                                  (filter
+                                   #(contains? @network-preference-names-state (:network-name %))))]
+        [:<>
+         [quo/drawer-top
+          {:title       (i18n/label :t/network-preferences)
+           :description (i18n/label :t/network-preferences-desc)}]
+         [quo/data-item
+          {:status          :default
+           :size            :default
+           :description     :default
+           :label           :none
+           :blur?           false
+           :card?           true
+           :title           (i18n/label :t/address)
+           :custom-subtitle (fn []
+                              [quo/address-text
+                               {:networks current-networks
+                                :address  address
+                                :format   :long}])
+           :container-style (merge style/data-item
+                                   {:background-color (colors/theme-colors colors/neutral-2_5
+                                                                           colors/neutral-90
+                                                                           theme)})}]
+
+         [quo/category
+          {:list-type :settings
+           :data      [(make-network-item mainnet
+                                          {:color               color
+                                           :network-preferences @network-preference-names-state
+                                           :on-change           #(toggle-network (:network-name
+                                                                                  mainnet))})]}]
+         [quo/category
+          {:list-type :settings
+           :label     (i18n/label :t/layer-2)
+           :data      (mapv #(make-network-item %
+                                                {:color               color
+                                                 :network-preferences @network-preference-names-state
+                                                 :on-change           (fn [_]
+                                                                        (toggle-network (:network-name
+                                                                                         %)))})
```

Comment:
Maybe better in terms of readability to swap the use of `(fn[])` with the use of `#()`, it's better to use `#()` for smaller functions.
i.e.:
```clojure
(mapv (fn [var] (make-network-item var ... :on-change #(toggle-network ...))
```